File: atcoder.jp/contests/abc159/tasks/abc159_b/main.py
S = input()


def is_palindrome(s: str) -> bool:
    return s == s[::-1]


def former(s: str) -> str:
    n = len(s)
    return s[: (n - 1 // 2)]


def latter(s: str) -> str:
    n = len(s)
    return s[(n + 3) // 2 - 1 :]


ok = all(map(is_palindrome, [S, former(S), latter(S)]))
ans = "Yes" if ok else "No"

print(ans)

File: atcoder.jp/contests/abc159/tasks/abc159_b/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];
const print = (...messages: any) => console.log(...messages);

const reverse = (s: string) => s.split("").reverse().join("");

const S = input();

const isPalindrome = (s: string) => s === reverse(s);
const former = (s: string) => s.slice(0, (s.length - 1) / 2);
const latter = (s: string) => s.slice((s.length + 3) / 2 - 1);

const ok = [S, former(S), latter(S)].every(isPalindrome);
const ans = ok ? "Yes" : "No";

print(ans);
